docs/design.md

```diff
@@ -0,0 +1,431 @@
+  types of metadata include a `name` field which must be unique for each
+  document `schema`.
+* `data` - Data to be validated by the schema described by the `schema`
+  field. Deckhand only interacts with content here as instructed to do so by
+  the `metadata` section. The form of this section is considered to be
+  completely owned by the `namespace` in the `schema`.
+
+#### Document Metadata
+
+There are 3 supported kinds of document metadata. Documents with `Document`
+metadata are the most common, and are used for normal configuration data.
+Documents with `Control` metadata are used to customize the behavior of
+Deckhand. Documents with `Tombstone` metadata are used to delete pre-existing
+documents with either `Document` or `Control` metadata.
+
+##### schema: metadata/Document/v1
+
+This type of metadata allows the following metadata hierarchy:
+
+* `name` - string, required - Unique within a revision for a given `schema`.
+* `storagePolicy` - string, required - Either `cleartext` or `encrypted`.
+* `layeringDefinition` - dict, required - Specifies
+  * `abstract` - boolean, required - An abstract document is not expected to
+    pass schema validation after layering and substitution are applied.
+    Non-abstract (concrete) documents are.
+  * `layer` - string, required - References a layer in the `LayeringPolicy`
+    control document.
+  * `parentSelector` - labels, optional - Used to construct document chains for
+    executing merges.  See the Layering section below for details.
+  * `actions` - list, optional - A sequence of actions to apply this documents
+    data during the merge process.
+    * `method` - string, required - How to layer this content.
+    * `path` - string, required - What content in this document to layer onto
+      parent content.
+* `substitutions` - list, optional - A sequence of substitutions to apply. See
+  the Substitutions section for additional details.
+  * `dest` - dict, required - A description of the inserted content destination.
+    * `path` - string, required - The JSON path where the data will be placed
+      into the `data` section of this document.
+  * `src` - dict, required - A description of the inserted content source.
+    * `schema` - string, required - The `schema` of the source document.
+    * `name` - string, required - The `metadata.name` of the source document.
+    * `path` - string, required - The JSON path from which to extract data in
+      the source document relative to its `data` section.
+
+Here is a fictitious example of a complete document which illustrates all the
+valid fields in the `metadata` section.
+
+```yaml
+---
+schema: some-service/ResourceType/v1
+metadata:
+  schema: metadata/Document/v1
+  name: unique-name-given-schema
+  storagePolicy: cleartext
+  labels:
+    genesis: enabled
+    master: enabled
+  layeringDefinition:
+    abstract: true
+    layer: region
+    parentSelector:
+      labels: parents
+      must: have
```

Comment:
Actually, "must" was mean to be sample data.  The idea is that `parentSelector` is a collection of key + value pairs that must all be present in the parents for them to be selected.

I'll chose different sample data to avoid confusion.  We'll also have to clarify the selector/label paradigm a bit more.